src/lib/schemas.ts

```diff
@@ -434,7 +434,26 @@ export const experimentFullNewSchema = experimentFullSchema.shape({
     .defined()
     .min(1, 'At least one metric assignment is required'),
   segmentAssignments: yup.array(segmentAssignmentNewSchema).defined(),
-  variations: yup.array<VariationNew>(variationNewSchema).defined().min(2),
+  variations: yup
+    .array<VariationNew>(variationNewSchema)
+    .defined()
+    .min(2)
+    .test(
+      'default-variation-exists',
+      'A default variation is required',
+      (variations: VariationNew[]) => variations && variations.some((variation) => variation.isDefault),
+    )
+    .test(
+      'max-total',
+      'The sum of allocated percentages must be less than or equal to 100',
+      (variations: VariationNew[]) =>
+        variations && variations.reduce((acc, variation) => acc + Number(variation.allocatedPercentage), 0) <= 100,
+    )
+    .test(
+      'unique-names',
+      'Variation names must be unique.',
```

Comment:
Consistency :slightly_smiling_face: 

```suggestion
      'Variation names must be unique',
```